sbt-remote-cache/src/main/scala/sbt/internal/GrpcActionCacheStore.scala

```diff
@@ -185,22 +187,19 @@ class GrpcActionCacheStore(
   /**
    * https://github.com/bazelbuild/remote-apis/blob/9ff14cecffe5287ba337f857731ceadfc2d80de9/build/bazel/remote/execution/v2/remote_execution.proto#L403
    */
-  override def syncBlobs(refs: Seq[HashedVirtualFileRef], outputDirectory: Path): Seq[Path] =
+  override def syncBlobs(refs: Seq[HashedVirtualFileRef]): Seq[Path] =
     val result = doGetBlobs(refs)
     val blobs = result.getResponsesList().asScala.toList
     val allOk = blobs.forall(_.getStatus().getCode() == 0)
     if allOk then
       // do not assume the responses to come in order
       val lookupResponse: Map[Digest, BatchReadBlobsResponse.Response] =
         Map(blobs.map(res => toDigest(res.getDigest) -> res): _*)
-      refs.map: r =>
-        val digest = Digest(r)
+      refs.map: ref =>
+        val digest = Digest(ref)
         val blob = lookupResponse(digest)
         val casFile = disk.putBlob(blob.getData().newInput(), digest)
-        val shortPath =
-          if r.id.startsWith("${OUT}/") then r.id.drop(7)
-          else r.id
-        val outPath = outputDirectory.resolve(shortPath)
+        val outPath = converter.toPath(ref)
```

Comment:
Wouldn't this allow remote files to be placed at an arbitrary location? The contract of the cache (remote or disk) should be that all files would be placed in some known CAS and symlink pointing to it.

Reply:
> Wouldn't this allow remote files to be placed at an arbitrary location?

Yes it would, and I see how it can be problematic, if the file is a symlink and it is committed to git.

But we don't have any guarantee that the declared output file of a task is in the out directory. Any user can redefine the `target` directory to wherever they want but the cache won't be able to restore it. It will create a non-sense `target/out/${BASE}/custom-target` instead.

So what should we do:
1. Prevent the cache from storing any file that is outside the output directory? Silently or with a warning?
2. Cache any file but if it is outside the `out` directory, we copy it instead of creating a symlink?

Reply:
Also if the `target` is customized, then maybe out should be customized to be under it anyway?

Reply:
> Also if the target is customized, then maybe out should be customized to be under it anyway?

When I say `target` I mean the `target` task of a project: `target/out/$platform/scala-$scalaVersion/$moduleName`

Reply:
> If possible we should try 1 I think, and fail the cache function.

Like we throw an exception? Or we just don't cache anything as if nothing happened.